### PR TITLE
Relabel image when selinux policy is installed

### PR DESCRIPTION
**Pull request by DaanDeMeyer**
If an selinux policy is installed in the image, let's relabel it automatically.

### CODE REVIEW

File: mkosi/__init__.py [hunk 1]
```diff
@@ -3135,6 +3135,25 @@ def run_preset_all(state: MkosiState) -> None:
         run(["systemctl", "--root", state.root, "preset-all"])
 
 
+def run_selinux_relabel(state: MkosiState) -> None:
+    selinux = state.root / "etc/selinux/config"
+    if not selinux.exists():
+        return
+
+    policy = run(["sh", "-c", f". {selinux} && echo $SELINUXTYPE"], text=True, stdout=subprocess.PIPE).stdout.strip()
```
Comment (behrmann) [inline mkosi/__init__.py:3143]: Wouldn't that be a good use of `shell=True`?

File: mkosi/__init__.py [hunk 2]
```diff
@@ -3135,6 +3135,25 @@ def run_preset_all(state: MkosiState) -> None:
...
+    if not policy:
+        return
+
+    fc = Path('/etc/selinux') / policy / 'contexts/files/file_contexts'
+
+    # We want to be able to relabel the underlying APIVFS mountpoints, so mount root non-recursive to a
+    # temporary location so that the underlying mountpoints become visible.
+    cmd = f"mkdir /tmp/relabel && mount --bind / /tmp/relabel && exec setfiles -m -r /tmp/relabel -F {fc} /tmp/relabel || exit $?"
```
Comment (behrmann) [inline mkosi/__init__.py:3151]:
```suggestion
    cmd = f"set -e; mkdir /tmp/relabel; mount --bind / /tmp/relabel; exec setfiles -m -r /tmp/relabel -F {fc} /tmp/relabel"
```
maybe easier as a oneliner to reason about, without having to think about associativity. (Only tested with bash invoked as `/bin/sh`.)

**Review by behrmann — COMMENTED** [review]
LGTM, two nits.

File: mkosi/__init__.py (hunk 1 above)
Comment (DaanDeMeyer) [reply mkosi/__init__.py:3143]: Maybe but I wouldn't sweat it with just one usecase

File: mkosi/__init__.py (hunk 2 above)
Comment (DaanDeMeyer) [reply mkosi/__init__.py:3151]: I prefer the above actually

**Comment by DaanDeMeyer** [comment]
Note that this doesn't work on its own, we also need https://github.com/systemd/systemd/pull/26410

File: mkosi/__init__.py (hunk 2 above)
Comment (behrmann) [reply mkosi/__init__.py:3151]: I just did another quick check and they are semantically different:
```
$ /bin/sh -c "set -e; foo() { return 19 ; }; echo foo; foo; echo bar"; echo $?
foo
19
$ /bin/sh -c "foo() { return 19 ; }; echo foo && foo && echo bar || exit $?"; echo $?
foo
0
```
So it does depend a bit on what you want.

**Pull request merged**